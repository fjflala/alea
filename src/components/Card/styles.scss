.ui-card {
  background: white;
  border-radius: 4px;
  padding: 32px;
  margin: 32px;
  box-shadow: 1px 1px 4px rgba($color: #333, $alpha: .4);
}