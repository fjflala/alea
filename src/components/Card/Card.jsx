/**
 * Module dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents a Card
 * @param {object} props - Props of the element
 * @param {string} props.className - Custom className for the card
 */
const Card = ({
  className,
  children,
}) => (
  <div className={`ui-card ${className}`}>
    {children}
  </div>
);

/**
 * PropTypes definitions
 */
Card.propTypes = {
  className: PropTypes.string,
};

Card.defaultProps = {
  className: '',
};

/**
 * Expose component
 */
export default Card;
