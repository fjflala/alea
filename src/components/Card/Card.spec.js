import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('<Card />', () => {
  it('should correctly render', () => {
    const props = {
      className: 'custom-class',
    };

    const tree = renderer
      .create(<Card {...props} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});