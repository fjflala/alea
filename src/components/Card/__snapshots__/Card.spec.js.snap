// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should correctly render 1`] = `
<div
  className="ui-card custom-class"
/>
`;
